client/signup/steps/woocommerce-install/confirm/index.tsx

```diff
@@ -59,11 +60,17 @@ const StyledNextButton = styled( NextButton )`
 	}
 `;
 
-function SupportLink() {
+function SupportLink( { domain }: { domain: string } ): ReactElement {
 	return (
 		<SupportLinkContainer>
 			{ createInterpolateElement( __( 'Need help? <a>Contact support</a>' ), {
-				a: <SupportLinkStyle href="/help/contact" />,
+				a: (
+					<SupportLinkStyle
+						href={ addQueryArgs( '/help/contact', {
+							redirect_to: `/start/woocommerce-install/confirm?site=${ domain }`,
```

Comment:
We might want to drop the site= part from this if we're going to enable flow resumes (https://github.com/Automattic/wp-calypso/pull/59185)